scipy/interpolate/_bspl.pyx

```diff
@@ -412,3 +413,52 @@ def _norm_eq_lsq(const double[::1] x,
                 # ... and A.T @ y
                 for ci in range(rhs.shape[1]):
                     rhs[row, ci] = rhs[row, ci] + wrk[r] * y[j, ci] * wval
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _make_design_matrix(const double[::1] x,
+                const double[::1] t,
+                int k,
+                kind):
+    '''
+    Returns a design matrix for BSpline object in "dense" or "CSR" formats.
+    
+    Parameters
+    ----------
+    x : 1-D array, shape(n,)
+        Values of x - coordinate of a given set of points (n >= 1).   
+    t : 1-D array, shape(nt,)
+        Vector of knots.
+    k : int
+        The maximum degree of spline.
```

Comment:
```suggestion
    x : ndarray, shape (n,)
        Points to evaluate the spline at.
    t : ndarray, shape (nt,)
        Sorted 1D array of knots.
    k : int
        Spline order.
```
What exactly is `nt`. I think the option for extrapolation is implicitly in it. Would be nice to make it more transparent.

Reply:
ndarray -> array_like